lib/interval/interval.ex

```diff
@@ -220,8 +249,14 @@ defmodule Timex.Interval do
       if has_recursion_ended?(current_date, end_date, right_open) do
         {:done, acc}
       else
-        next_date = Timex.shift(current_date, keywords)
-        do_reduce({next_date, end_date, right_open, keywords}, fun.(current_date, acc), fun)
+        case Timex.shift(current_date, keywords) do
+          {:error, {:unknown_shift_unit, :invalid_step}} ->
```

Comment:
Duh. I forgot that :invalid_step was what I was testing for, and assumed it was just part of the error that Timex.shift regularly returns. Great catch.